eskip/eskip.go

```diff
@@ -712,29 +712,28 @@ func ParsePredicates(p string) ([]*Predicate, error) {
 	return ps, nil
 }
 
-const randomIdLength = 16
-
-var routeIdRx = regexp.MustCompile(`\W`)
+const (
+	randomIdLength = 16
+	// does not contain underscore to produce compatible output with previously used flow id generator
+	alphabet = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+)
 
 // generate weak random id for a route if
 // it doesn't have one.
+//
+// Deprecated: do not use, generate valid route id that matches [a-zA-Z_] yourself.
```

Comment:
deprecate in the sense of stable?
If so, why do we need to deprecate it?
If we do not plan to delete it, then it's just fine and we do not need to apply the ignore SA1019

Reply:
To hint lib users to avoid it.
I'd also delete it in the future as it has strange API contract.

Or do you propose that we keep it or replace with something else?

Reply:
I just wanted to ask what the plan is, because the PR description nor commit does show what the plan is.
And you talked about deprecated is good because it means it is stable in the last days.